ui/app/AppLayouts/Wallet/stores/RootStore.qml

```diff
@@ -309,33 +309,27 @@ QtObject {
 
     function getAssetForSendTx(tx) {
         if (tx.isNFT) {
-            return {
-                uid: tx.tokenID,
-                chainId: tx.chainId,
-                name: tx.nftName,
-                imageUrl: tx.nftImageUrl,
-                collectionUid: "",
-                collectionName: ""
-            }
-        } else {
-            return tx.symbol
+            return collectiblesStore.getUidForData(tx.tokenID, tx.tokenAddress, tx.chainId)
         }
+        return tx.symbol
     }
 
     function isTxRepeatable(tx) {
```

Comment:
Just thinking out loud.. don't we have a `TransactionsStore.qml`? Would this be better there, maybe?

Reply:
That's just a thought to keep in mind when we start redefining stores... For me, a store should contain backend references to specific contexts, and not related to views, that is, if we have a `TransactionsStore` and we need to operate with transactions inside of `SendModal` but also in `ActivityView`, this is okay. For me, _RootStore_ should "just" instantiate the entire list of "wallet context-specific stores" and maybe have some calls to some generic backend methods or properties, nothing more!

Reply:
Yeah, I agree with @noeliaSD here. Stores should not be coupled with specific views. They should aggregate models and actions from given domain, like e.g. related to transactions, no matter where it's used.